File: src/main/java/mediathek/gui/actions/SearchProgramUpdateAction.java
package mediathek.gui.actions;

import mediathek.update.ProgrammUpdateSuchen;

import javax.swing.*;
import java.awt.event.ActionEvent;

public class SearchProgramUpdateAction extends AbstractAction {
    public SearchProgramUpdateAction() {
        putValue(Action.NAME, "Nach Update suchen...");
    }

    @Override
    public void actionPerformed(ActionEvent e) {
        new ProgrammUpdateSuchen().checkVersion(true, false, false, false);
    }
}

File: src/main/kotlin/mediathek/gui/actions/SearchProgramUpdateAction.kt
package mediathek.gui.actions

import mediathek.update.ProgrammUpdateSuchen
import java.awt.event.ActionEvent
import javax.swing.AbstractAction
import javax.swing.JFrame

class SearchProgramUpdateAction(
    private val parent: JFrame,
) : AbstractAction() {
    init {
        putValue(NAME, "Nach Update suchen...")
    }

    override fun actionPerformed(event: ActionEvent?) {
        ProgrammUpdateSuchen(ownerProvider = { parent }).checkVersion(
            showAlert = true,
            showProgramInformation = false,
            showAllInformation = false,
            silent = false,
        )
    }
}
